wire/net/ioconn.go

```diff
@@ -39,16 +41,16 @@ func NewIoConn(conn io.ReadWriteCloser) Conn {
 }
 
 func (c *ioConn) Send(e *wire.Envelope) error {
-	if err := e.Encode(c.conn); err != nil {
+	if err := wire.EncodeEnvelope(c.conn, *e); err != nil {
 		c.conn.Close()
 		return err
 	}
 	return nil
 }
 
 func (c *ioConn) Recv() (*wire.Envelope, error) {
-	var e wire.Envelope
-	if err := e.Decode(c.conn); err != nil {
+	e, err := wire.DecodeEnvelope(c.conn)
```

Comment:
I would prefer to pass around Envelopes as values, because we do not need a reference to it anywhere.

However, since it is passed as pointer in many other places, I will update this functions also to use a pointer, for consistency.

Reply:
Updated in 1b743c17, 10ec67cd, 338dbafd.